docs/architecture/adr-013-metrics.md

```diff
@@ -0,0 +1,90 @@
+# ADR 013: Observability
+
+## Changelog
+
+- 20-01-2020: Initial Draft
+
+## Status
+
+Proposed
+
+## Context
+
+There has been discussion around exposing more metrics to users and node operators about the application. Currently there is only a way to expose metrics from Tendermint and not the application itself. To bring more visibility into applications, I would like to propose reporting of metrics through [Prometheus](https://prometheus.io/).
+
+Extending `AppModuleBasic` to support registering of metrics would enable developers to see more information about individual modules.
+
+```go
+type AppModuleBasic interface {
+  Name() string
+  RegisterCodec(*codec.Codec)
+  RegisterMetrics(namespace string, labelsAndValues ...string) *Metrics
+
+  // genesis
+  DefaultGenesis() json.RawMessage
+  ValidateGenesis(json.RawMessage) error
+
+  // client functionality
+  RegisterRESTRoutes(context.CLIContext, *mux.Router)
+  GetTxCmd(*codec.Codec) *cobra.Command
+  GetQueryCmd(*codec.Codec) *cobra.Command
+}
+// .....
+
+func (bm BasicManager) RegisterMetrics(appName) MetricsProvider {
+	for _, b := range bm {
+		b.RegisterMetrics(appName, labelsAndValues)
+	}
+}
+```
+
+Each module can define its own `CreateMetrics` function:
```

Comment:
Metrics should be created within a new package `x/<module>/observability/metrics.go`. Please include that in your ADR and update the module spec